firedrake/variational_solver.py

```diff
@@ -133,7 +133,7 @@ class NonlinearVariationalSolver(OptionsManager, NonlinearVariationalSolverMixin
     DEFAULT_SNES_PARAMETERS = DEFAULT_SNES_PARAMETERS
 
     # Looser default tolerance for KSP inside SNES.
-    DEFAULT_KSP_PARAMETERS = MappingProxyType(DEFAULT_KSP_PARAMETERS | {'ksp_rtol': 1e-5})
+    DEFAULT_KSP_PARAMETERS = MappingProxyType(DEFAULT_KSP_PARAMETERS.update({'ksp_rtol': 1e-5}))
```

Comment:
Yeah, this doesn't work for two reasons:

1. update is in-place
2. update returns `None` because python hates function chaining.

Reply:
And 3. `MappingProxyType` doesn't even have an update method!